src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TypeImportCompletionProviderTests.cs

```diff
@@ -907,6 +907,270 @@ class Bat
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task CommitAttributeTypeInAttributeNameContext(SourceCodeKind kind)
+        {
+            var file1 = @"
+namespace Foo
+{
+    public class MyAttribute : System.Attribute { }
+}";
+
+            var file2 = @"
+namespace Test
+{
+    [$$
+    class Program { }
+}";
+
+            string expectedCodeAfterCommit = @"
+using Foo;
+
+namespace Test
+{
+    [My$$
+    class Program { }
+}";
+
+            var markup = CreateMarkupForSingleProject(file2, file1, LanguageNames.CSharp);
+            await VerifyCustomCommitProviderAsync(markup, "My", expectedCodeAfterCommit, sourceCodeKind: kind);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task AttributeTypeInNonAttributeNameContext()
+        {
+            var file1 = @"
+namespace Foo
+{
+    public class MyAttribute : System.Attribute { }
+    public class MyClass { }
+}";
+
+            var file2 = @"
+namespace Test
+{
+    class Program 
+    {
+        $$
+    }
+}";
+            var markup = CreateMarkupForSingleProject(file2, file1, LanguageNames.CSharp);
+
+            await VerifyTypeImportItemExistsAsync(markup, "MyAttribute", glyph: (int)Glyph.ClassPublic, inlineDescription: "Foo", expectedDescriptionOrNull: "class Foo.MyAttribute");
+            await VerifyTypeImportItemIsAbsentAsync(markup, "My", inlineDescription: "Foo");
+            await VerifyTypeImportItemExistsAsync(markup, "MyClass", glyph: (int)Glyph.ClassPublic, inlineDescription: "Foo", expectedDescriptionOrNull: "class Foo.MyClass");
+        }
+
+        [InlineData(SourceCodeKind.Regular)]
+        [InlineData(SourceCodeKind.Script)]
+        [WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task CommitAttributeTypeInNonAttributeNameContext(SourceCodeKind kind)
+        {
+            var file1 = @"
+namespace Foo
+{
+    public class MyAttribute : System.Attribute { }
+}";
+
+            var file2 = @"
+namespace Test
+{
+    class Program 
+    {
+        $$
+    }
+}";
+
+            string expectedCodeAfterCommit = @"
+using Foo;
+
+namespace Test
+{
+    class Program 
+    {
+        MyAttribute$$
+    }
+}";
+            var markup = CreateMarkupForSingleProject(file2, file1, LanguageNames.CSharp);
+            await VerifyCustomCommitProviderAsync(markup, "MyAttribute", expectedCodeAfterCommit, sourceCodeKind: kind);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task AttributeTypeWithoutSuffixInAttributeNameContext()
+        {
+            var file1 = @"
+namespace Foo
+{
+    public class Myattribute : System.Attribute { }
```

Comment:
The difference here was hard for me to spot at first (the lowercase `a`). Maybe `AttributeTypeWithIncorrectlyCasedSuffixInAttributeNameContext` or a comment?

Reply:
Oh, kinda like the comments in the analogous VB tests.

Reply:
OK, added similar comments in C# tests